components/logins/android/src/main/java/mozilla/appservices/logins/DatabaseLoginsStorage.kt

```diff
@@ -142,68 +163,100 @@ class DatabaseLoginsStorage(private val dbPath: String) : AutoCloseable, LoginsS
+                    PasswordSyncAdapter.INSTANCE.sync15_passwords_get_by_id(raw, id, error)
+                }
+            }?.getAndConsumeRustString()
+            json?.let { ServerPassword.fromJSON(it) }
+        }
     }
 
     @Throws(LoginsStorageException::class)
     override fun touch(id: String) {
-        rustCallWithLock { raw, error ->
-            PasswordSyncAdapter.INSTANCE.sync15_passwords_touch(raw, id, error)
+        writeQueryCounters.measure {
+            rustCallWithLock { raw, error ->
+                LoginsStoreMetrics.writeQueryTime.measure {
+                    PasswordSyncAdapter.INSTANCE.sync15_passwords_touch(raw, id, error)
+                }
+            }
         }
     }
 
     @Throws(LoginsStorageException::class)
     override fun list(): List<ServerPassword> {
-        val json = rustCallWithLock { raw, error ->
-            PasswordSyncAdapter.INSTANCE.sync15_passwords_get_all(raw, error)
-        }.getAndConsumeRustString()
-        return ServerPassword.fromJSONArray(json)
+        return readQueryCounters.measure {
+            val json = rustCallWithLock { raw, error ->
+                LoginsStoreMetrics.readQueryTime.measure {
+                    PasswordSyncAdapter.INSTANCE.sync15_passwords_get_all(raw, error)
+                }
+            }.getAndConsumeRustString()
+            ServerPassword.fromJSONArray(json)
+        }
     }
 
     @Throws(LoginsStorageException::class)
     override fun getByHostname(hostname: String): List<ServerPassword> {
-        val json = rustCallWithLock { raw, error ->
-            PasswordSyncAdapter.INSTANCE.sync15_passwords_get_by_hostname(raw, hostname, error)
-        }.getAndConsumeRustString()
-        return ServerPassword.fromJSONArray(json)
+        return readQueryCounters.measure {
+            val json = rustCallWithLock { raw, error ->
+                LoginsStoreMetrics.readQueryTime.measure {
+                    PasswordSyncAdapter.INSTANCE.sync15_passwords_get_by_hostname(raw, hostname, error)
+                }
+            }.getAndConsumeRustString()
+            ServerPassword.fromJSONArray(json)
+        }
     }
 
     @Throws(LoginsStorageException::class)
     override fun add(login: ServerPassword): String {
-        val s = login.toJSON().toString()
-        return rustCallWithLock { raw, error ->
-            PasswordSyncAdapter.INSTANCE.sync15_passwords_add(raw, s, error)
-        }.getAndConsumeRustString()
+        return writeQueryCounters.measure {
+            val s = login.toJSON().toString()
+            rustCallWithLock { raw, error ->
+                LoginsStoreMetrics.writeQueryTime.measure {
+                    PasswordSyncAdapter.INSTANCE.sync15_passwords_add(raw, s, error)
+                }
+            }.getAndConsumeRustString()
+        }
     }
 
     @Throws(LoginsStorageException::class)
     override fun importLogins(logins: Array<ServerPassword>): Long {
```

Comment:
This function performs substantially more writes than any other, and would likely skew the timings. We might want to use a different counter tthan the same as the other write counters IMO.